resources/assets/components/CampaignSingle/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { flatMap, keyBy, map, sample, forEach, reject, filter } from 'lodash';
+
+import InboxItem from '../InboxItem';
+import ModalContainer from '../ModalContainer';
+import HistoryModal from '../HistoryModal';
+import PostFilter from '../PostFilter';
+import StatusCounter from '../StatusCounter';
+import { RestApiClient} from '@dosomething/gateway';
+
+class CampaignSingle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const posts = keyBy(flatMap(props.signups, signup => {
+      return signup.posts.map(post => {
+        post.signup = signup;
+        return post;
+      });
+    }), 'id');
+
+    this.state = {
+      posts: posts,
+      filteredPosts: posts,
+      postTotals: props.post_totals,
+      displayHistoryModal: false,
+      historyModalId: null,
+    };
+
+    this.filterPosts = this.filterPosts.bind(this);
+
+    this.api = new RestApiClient;
+    this.updateQuantity = this.updateQuantity.bind(this);
+    this.showHistory = this.showHistory.bind(this);
+    this.hideHistory = this.hideHistory.bind(this);
+  }
+
+  filterPosts(status) {
+    const posts = filter(this.state.posts, {'status' : status.toLowerCase()});
+
+    this.setState({ filteredPosts: posts });
+  }
+
+  // Open the history modal of the given post
+  // @TODO - Move this into the InboxItem component so
+  // we don't have to duplicate this functionality when every we want to use Inbox Items.
+  showHistory(postId, event) {
+    event.preventDefault()
+
+    this.setState({
+      displayHistoryModal: true,
+      historyModalId: postId,
+    });
+  }
+
+  // Close the open history modal
+  hideHistory(event) {
+    if (event) {
+      event.preventDefault();
+    }
+
+    this.setState({
+      displayHistoryModal: false,
+      historyModalId: null,
+    });
+  }
+
+  // Update a signups quanity.
+  updateQuantity(post, newQuantity) {
+    // Fields to send to /posts
+    const fields = {
+      northstar_id: post.user.id,
+      campaign_id: post.signup.campaign_id,
+      campaign_run_id: post.signup.campaign_run_id,
+      quantity: newQuantity,
+    };
+
+    // Make API request to Rogue to update the quantity on the backend
+    let response = this.api.post('api/v2/posts', fields);
```

Comment:
I'm a little wary of calling the promise here `response`, since that implies it's the actual HTTP response. Maybe `request` instead? Or just chain the `.then()` off the original call rather than storing it in a variable?